### PR TITLE
EVG-16367 Add canCreateProject check to the createProject Modal

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-16367](https://jira.mongodb.org/browse/EVG-16367)

### Description 
This restricts the createProject button and modal to people who have the right permissions.

### CODE REVIEW

File: src/components/ProjectSettingsTabs/CreateProjectModal.tsx [hunk 1]
```diff
@@ -70,7 +72,16 @@ interface Props {
...
-  const isAdmin = true; // todo after EVG-16353
+
+  // USER PERMISSIONS QUERY
+  const { data: userPermissionsData } = useQuery<GetUserPermissionsQuery>(
+    GET_USER_PERMISSIONS
+  );
+  const { user } = userPermissionsData || {};
+  const { permissions } = user || {};
+  const canCreateProject = permissions?.canCreateProject;
+  console.log("userPermissionsData: ", userPermissionsData);
```
Comment (khelif96) [inline src/components/ProjectSettingsTabs/CreateProjectModal.tsx:83]: console.log

File: src/components/ProjectSettingsTabs/CreateProjectModal.tsx [hunk 2]
```diff
@@ -109,13 +120,13 @@ export const CreateProjectModal: React.FC<Props> = ({ owner, repo }) => {
     refetchQueries: ["ProjectSettings", "RepoSettings"],
   });
 
-  if (!isAdmin) {
+  if (!canCreateProject) {
```
Comment (khelif96) [inline src/components/ProjectSettingsTabs/CreateProjectModal.tsx:123]: can we fallback to something?

File: src/gql/queries/get-user-permissions.graphql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+query GetUserPermissions {
+  user {
+    userId
+    permissions {
+      canCreateProject
+      userId
```
Comment (khelif96) [inline src/gql/queries/get-user-permissions.graphql:6]: userId isn't needed here.

File: src/components/ProjectSettingsTabs/CreateProjectModal.tsx (hunk 2 above)
Comment (malikchaya2) [reply src/components/ProjectSettingsTabs/CreateProjectModal.tsx:123]: Can you please elaborate a bit more? What kind of fallback? If someone doesn't have permissions, we don't want them to see the button. Do you mean like an empty div or something?

File: src/components/ProjectSettingsTabs/CreateProjectModal.tsx (hunk 1 above)
Comment (malikchaya2) [reply src/components/ProjectSettingsTabs/CreateProjectModal.tsx:83]: thanks!

File: src/components/ProjectSettingsTabs/CreateProjectModal.tsx (hunk 2 above)
Comment (khelif96) [reply src/components/ProjectSettingsTabs/CreateProjectModal.tsx:122]: Sorry i missed the context of this change i didn't realize this was for the button that toggles the modal.

**Review by khelif96 — APPROVED** [review]

**Comment by ablack12** [comment]
Not reviewing since there isn't a visual change to check on but let me know if that changes.

File: src/gql/generated/types.ts [hunk 4]
```diff
@@ -3909,6 +3909,15 @@ export type GetUserConfigQuery = {
   }>;
 };
 
+export type GetUserPermissionsQueryVariables = Exact<{ [key: string]: never }>;
+
+export type GetUserPermissionsQuery = {
+  user: {
+    userId: string;
+    permissions: { canCreateProject: boolean; userId: string };
```
Comment (sophstad) [inline src/gql/generated/types.ts:3917]: Looks like `yarn codegen` needs to be run again; this `userId` field was removed from the query.

**Review by sophstad — CHANGES_REQUESTED** [review]

File: src/gql/generated/types.ts [hunk 5]
```diff
@@ -3912,10 +3912,7 @@ export type GetUserConfigQuery = {
 export type GetUserPermissionsQueryVariables = Exact<{ [key: string]: never }>;
 
 export type GetUserPermissionsQuery = {
-  user: {
-    userId: string;
```
Comment (sophstad) [inline src/gql/generated/types.ts]: I think this `userId` should stay to help with Apollo caching (maybe @khelif96 can confirm)

Comment (khelif96) [reply src/gql/generated/types.ts]: Yes userId should stay for the user object. My earlier comment was only in reference to the permissions object.

Comment (malikchaya2) [reply src/gql/generated/types.ts]: restored

**Review by sophstad — APPROVED** [review]

**Pull request merged**